Dockerfile

```diff
@@ -1,15 +1,49 @@
+# LAYER builder — — — — — — — — — — — — — — — — — — — — — — —
 FROM golang:1.12-alpine AS builder
 ADD go.mod go.sum /app/
 WORKDIR /app/
 RUN apk --no-cache add git upx
 RUN go mod download
 ADD . /app/
 RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags "-X main.version=$(git describe --tags --abbrev=0)" -a -o /trivy cmd/trivy/main.go
-RUN upx --lzma --best /trivy
 
-FROM alpine:3.9
-RUN apk --no-cache add ca-certificates git
+
+# LAYER compress — — — — — — — — — — — — — — — — — — — — — — —
+FROM alpine:3.9 AS compress
 COPY --from=builder /trivy /usr/local/bin/trivy
-RUN chmod +x /usr/local/bin/trivy
 
-ENTRYPOINT ["trivy"]
+RUN set -eux && apk --update --no-cache add \
+    upx="3.95-r1" && \
```

Comment:
Hardcoding version numbers is nice but I would make it a little bit less strict.

Reply:
No problem.
On my critical apps in prod, I prefer to do this as I had one nassssty bug cause but a package in the past.

Reply:
done